dask/bytes/pyarrow.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import print_function, division, absolute_import
+
+import re
+import fnmatch
+import posixpath
+
+from ..base import tokenize
+
+import pyarrow as pa
+
+
+class HDFS3Wrapper(pa.filesystem.DaskFileSystem):
+    """Pyarrow compatibility wrapper class"""
+    def isdir(self, path):
+        return self.fs.isdir(path)
+
+    def isfile(self, path):
+        return self.fs.isfile(path)
+
+
+_MIN_PYARROW_VERSION_SUPPORTED = '0.8.1.dev81'
+
+
+class PyArrowHadoopFileSystem(object):
+    sep = "/"
+
+    def __init__(self, **kwargs):
+        self.fs = pa.hdfs.HadoopFileSystem(**kwargs)
+
+    @classmethod
+    def from_pyarrow(cls, fs):
+        out = object.__new__(cls)
+        out.fs = fs
+        return out
+
+    def open(self, path, mode='rb', **kwargs):
+        return self.fs.open(path, mode=mode, **kwargs)
+
+    def glob(self, path):
+        return sorted(_glob(self.fs, path))
+
+    def mkdirs(self, path):
+        return self.fs.mkdir(path)
```

Comment:
This will fail if the parent directory does not exist, I think - unlike the usual understanding of `mkdirs`.

Reply:
`libhdfs` always makes intermediate directories. I added an (ignored) keyword that arrow supports that should better document the intention here.